sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -505,10 +505,14 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
   protected def visitStringConstant(
```

Comment:
does it work for `ADD PARTITION(a=date'xxx')`, etc.?

Reply:
> does it work for `ADD PARTITION(a=date'xxx')`, etc.?

Support and add test case  in UT